XMLCreator/XMLCreator.cpp

```diff
@@ -0,0 +1,292 @@
+	if (!current->children.empty())
+	{
+		file_name << character.NEW_LINE;
+		depth++;
+		for (size_t i = 0; i < current->children.size(); i++)
+		{
+			WriteTreeToFile(current->children[i], depth, file_name, character);
+		}
+		--depth;
+		for (size_t j = 0; j < depth; j++)
+		{
+			file_name << character.TAB;
+		}
+	}
+	file_name << character.END_TAG << current->tag << '>' << character.NEW_LINE;
+}
+
+bool XMLCreator::CreateStandartTree()
+{
+	if (m_root != nullptr)
+	{
+		return false;
+	}
+
+	const std::string LISTENER_PORT = std::to_string(SERVER_LISTENER_PORT);
+	const std::string TIMEOUT = std::to_string(SOCKET_TIMEOUT);
+	const std::string LEVEL = std::to_string(LOG_LEVEL);
+	const std::string INTERVAL = std::to_string(THREAD_INTERVAL);
+	const std::string WORKING_THREADS = std::to_string(MAX_WORKING_THREADS);
+	const std::string BLOCKING = "0";
+
+	const int ROOT_TAGS_SIZE = 5;
+	const int SERVER_TAGS_SIZE = 4;
+	const int COMMUNICATION_TAGS_SIZE = 2;
+	const int LOGGING_TAGS_SIZE = 3;
+	std::array<std::string, ROOT_TAGS_SIZE> root_tags{ ATTR_SERVER, ATTR_COMMUNICATION_SETTINGS,
+										ATTR_LOGGING, ATTR_TIME, ATTR_THREAD_POOL };
+	std::array<std::string, SERVER_TAGS_SIZE> server_tags{ ATTR_SERVER_NAME, ATTR_SERVER_DISPLAY_NAME,
+										ATTR_SERVER_LISTENER_PORT, ATTR_SERVER_IP_ADDRESS };
+	std::array<std::string, SERVER_TAGS_SIZE> server_values{ SERVER_NAME, SERVER_DISPLAY_NAME, LISTENER_PORT,
+											SERVER_IP_ADDRESS };
+	std::array<std::string, COMMUNICATION_TAGS_SIZE> communication_tags{ ATTR_SOCKET_BLOCKING, ATTR_SOCKET_TIMEOUT };
+	std::array<std::string, COMMUNICATION_TAGS_SIZE> communication_values{ BLOCKING, TIMEOUT };
+	std::array<std::string, LOGGING_TAGS_SIZE> logging_tags{ ATTR_LOG_FILENAME, ATTR_LOG_LEVEL, ATTR_LOG_FLUSH };
+	std::array<std::string, LOGGING_TAGS_SIZE> logging_values{ LOG_FILENAME, LEVEL, BLOCKING };
+
+	std::shared_ptr<XMLNode> parent_node = nullptr;
+	std::shared_ptr<XMLNode> new_node = nullptr;
+	std::string node_value;
+	CreateElement(parent_node, ATTR_ROOT, node_value);
+	m_root = parent_node;
+	for (int i = 0; i < ROOT_TAGS_SIZE; i++)
+	{
+		CreateElement(new_node, root_tags[i], node_value);
+		ChildAppend(parent_node, new_node);
+	}
+	enum class Children
+	{
+		SERVER = 0, COMMUNICATION, LOGGING, TIME, THREADPOOL
+	};
+	if (m_root == nullptr)
+	{
+		return false;
+	}
+	parent_node = m_root->children[(int)Children::SERVER];
+	for (size_t i = 0; i < SERVER_TAGS_SIZE; i++)
+	{
+		CreateElement(new_node, server_tags[i], server_values[i]);
+		ChildAppend(parent_node, new_node);
+	}
+	parent_node = m_root->children[(int)Children::COMMUNICATION];
+	for (size_t i = 0; i < COMMUNICATION_TAGS_SIZE; i++)
+	{
+		CreateElement(new_node, communication_tags[i], communication_values[i]);
+		ChildAppend(parent_node, new_node);
+	}
+	parent_node = m_root->children[(int)Children::LOGGING];
+	for (size_t i = 0; i < LOGGING_TAGS_SIZE; i++)
```

Comment:
Maybe there're a better way to create one 'for' loop for all this cases. They are almost the same.

Reply:
CreateStandartTree() method was optimized.